### ISSUE TITLE
Unexpected duplicate selector: throwing false positives with special characters (scss)

### ISSUE THREAD

**Issue by harry-whorlow** [stylelint/stylelint#6116]
### What steps are needed to reproduce the bug?

Using special characters as annotation for selectors will cause StyleLint to throw an `Unexpected duplicate selector ""` error. Below you will find a code snippet with the problematic code

```
%border-right {
    border-right: 1px solid var(--light-blue-gray-200);
}

%padding-right {
    padding-right: 2%;
}

%sub-line-padding {
    padding-top: 5px;
}
```

As you can see we're using % as annotation for our selectors, however when running StyleLint it'll treat the % as the selector... and throw a duplicated selector error for each and every selector using the % key. This was working with previous versions of Stylelint however after the recent update to v14.8.4 it began throwing the error.

```
  8:1  ✖  Unexpected duplicate selector "", first used at line 4  no-duplicate-selectors
 12:1  ✖  Unexpected duplicate selector "", first used at line 4  no-duplicate-selectors
 24:1  ✖  Unexpected duplicate selector "", first used at line 4  no-duplicate-selectors
```

This error shows up on multiple files, all that have used the % key as annotation for the selector. 

### What Stylelint configuration is needed to reproduce the bug?

These should be unrelated, but heres my config.

```
"rules": {
        "at-rule-no-unknown": null,
        "scss/at-rule-no-unknown": true,
        "indentation": 4,
        "max-nesting-depth": 3,
        "selector-type-no-unknown": null,
        "selector-pseudo-element-no-unknown": null,
        "font-family-no-missing-generic-family-keyword": null,
        "no-empty-source": null,
        "font-family-name-quotes": null,
        "function-no-unknown": null,
    }
```

### How did you run Stylelint?

CLI: npm run lint:css

### Which version of Stylelint are you using?

14.8.4

### What did you expect to happen?

No "Unexpected duplicate selector" errors to be thrown.

### What actually happened?

` Unexpected duplicate selector ""` thrown for each and every selector using % special character.

![Screenshot 2022-05-25 at 14 38 29 (2)](https://user-images.githubusercontent.com/79278353/170263805-9d1767cf-aa6f-40ff-bbdd-f286d54c6d82.png)

As stated previously, this issues is thrown for all files using % as annotation for the selector.

### Does the bug relate to non-standard syntax?

Yes, it relates to SCSS

**Comment by JounQin** [comment]
Please upgrade to 14.8.5
